### PR TITLE
Prepare for store bookmarks demod+bw

### PR DESCRIPTION
**Pull request by jedi7**
* Bookmarks are now capable of storing demodulation and bandwidth
  (as low cut frequency and high cut frequency)
* WARNING: setting of bandwidht and demod not working now

Hi I need a little help with propagate of the selected bandwidht (low, high freq cut) to the gui.
Thanks :)

### CODE REVIEW

File: BookmarkInfo.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#ifndef BOOKMARK_INFO_H
```
Comment (BatchDrake) [inline BookmarkInfo.h:1]: Include header with licensing + author

File: Waterfall.cpp [hunk 2]
```diff
@@ -805,8 +807,27 @@ void Waterfall::mousePressEvent(QMouseEvent * event)
...
+                        << ", modulation: " << info.modulation;
+
+                    if(!info.modulation.isEmpty()) {
+                        emit newModulation(info.modulation);
+                    }
+
+                    m_DemodCenterFreq = info.frequency;
                     emit newDemodFreq(m_DemodCenterFreq, m_DemodCenterFreq - m_CenterFreq);
+
+                    if(info.lowFreqCut != 0 && info.highFreqCut != 0) {
```
Comment (BatchDrake) [inline Waterfall.cpp:825]: What happens with SSB modulations? In USB, info.highFreqCut = 0 and in LSB, info.lowFreqCut = 0

**Review by BatchDrake — CHANGES_REQUESTED** [review]
Apart from the SSB thing which maybe needs a bit more of clarification, the abstraction looks good so far. 

PS: sub-minor (coding style): I usually leave a space between keywords and parentheses (`if (...)`, `sizeof (...)`, `while (...)`, `switch (...)`) but not between function names, methods and macros with arguments (just in case you see me changing your code right after the merge). I believe I should put a coding style document somewhere.

File: Waterfall.cpp (hunk 2 above)
Comment (jedi7) [reply Waterfall.cpp]: changed to bandwidth(), thanks :)

File: BookmarkInfo.h (hunk 1 above)
Comment (jedi7) [reply BookmarkInfo.h:20]: fixed

**Comment by jedi7** [comment]
fixed code style. Maybe clang-format can be used to automatically format the code?

**Review by BatchDrake — APPROVED** [review]

**Pull request merged**